samcli/commands/init/__init__.py

```diff
@@ -126,12 +149,25 @@ def do_cli(
     # check for required parameters
     if location or (name and runtime and dependency_manager and app_template):
         # need to turn app_template into a location before we generate
-        extra_context = None
         if app_template:
             templates = InitTemplates(no_interactive, auto_clone)
             location = templates.location_from_app_template(runtime, dependency_manager, app_template)
             no_input = True
-            extra_context = {"project_name": name, "runtime": runtime}
+            default_context = {"project_name": name, "runtime": runtime}
+            if extra_context is None:
```

Comment:
This is all pretty dense. Why can't we do something like:

```
default_context = {"project_name": name, "runtime": runtime}
extra_context_dict = {}
try:
    extra_context_dict = json.loads(extra_context)
except JSONDecodeError:
   raise

default_context = {**default_context , **extra_context_dict}
```

Reply:
I was trying to avoid override for default parameter by switching order this does the work.